radixdlt/src/main/java/com/radixdlt/consensus/ProposerElectionFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.radixdlt.consensus.liveness.ProposerElection;
+import com.radixdlt.crypto.ECPublicKey;
+
+import java.util.Collection;
+
+public interface ProposerElectionFactory {
+	ProposerElection create(Collection<ECPublicKey> proposers);

```

Comment:
This should be an ImmutableList since order of proposers will matter given that proposers is something serialized and sent over the network.

Reply:
Good point on the list, will change that. Would rather avoid bringing ImmutableList into this, can always just make an immutable copy internally.